src/test/java/org/mockito/internal/matchers/CapturingMatcherTest.java

```diff
@@ -52,4 +54,20 @@ public void should_scream_when_nothing_yet_captured() throws Exception {
             fail();
         } catch (MockitoException e) {}
     }
+
+    @Test
+    public void should_not_fail_when_used_in_concurrent_tests() throws Exception {
+        //given
+        final CapturingMatcher<String> m = new CapturingMatcher<String>();
+
+        //when
+        m.captureFrom("concurrent access");
+        Iterator<String> iterator = m.getAllValues().iterator();
+        m.captureFrom("concurrent access");
+
+        //then
+        assertThat(iterator.hasNext()).isTrue();
+        assertThat(iterator.next()).isEqualTo("concurrent access"); // Potential ConcurrentModificationException
```

Comment:
Is it certain that without this fix the Exception is always thrown? If not, it would be nice to make this deterministic (although that probably is hard to achieve :cry: )